### DISCUSSION TITLE
Underline heading and numbering withoug gap

### DISCUSSION THREAD

**Discussion by CasparNuel** [typst/typst#4389] [Q&A]
I am trying to underline a heading with a numbering prefix.

This is my current setup:

```typst
#set heading(numbering: "1.1 ")

#show heading: underline

= This should be underlined
```

This results in the following output:

![Screenshot_20240613_170821_Samsung Internet](https://github.com/typst/typst/assets/73889285/20f08dc8-2f47-44e8-b984-13c5e6318521)

I would like to avoid the gap between the numbering and the heading. How can I achieve this? Both the option `evade` and the option `extent` for the underline command don't really do what I want.

Any input is greatly appreciated!

**Comment by CDucloux** [comment]
Instead of underlining, why not use the `line` function ? For instance, `line(length: 100%, stroke: gray)`

```typst
show heading: it => block(width: 100%)[
    #text(it, size: 18pt)
    #v(-0.3cm)
    #line(length: 100%, stroke: gray)
    #v(0.3cm)
  ]
```

Would give you an underlined heading (feel free to adjust the size of the text at will and the vertical space).

**Reply by dafitt** [reply]
Alternative with a #stack

```typst
#show heading: it => stack(
  spacing: 0.2%,
  text(it),
  line(length: 100%, stroke: 0.5pt),
)
```

**Reply by tovrstra** [reply]
Another alternative with `#box`, inspired by the solution of @daffit:

```typ
#show heading: it => box(
  stroke: (bottom: 0.5pt),
  inset: (bottom: 0.3em),
)[#it]
```

This puts a single line under all text but not past the text (It does not continue all the way to the right of the page.)
